### DISCUSSION TITLE
2 way audio still not working with reolink plugin

### DISCUSSION THREAD

**Discussion by GravityRZ** [koush/scrypted#1115] [Q&A] [answered]
running the latest firmware and trhe latest version of scrypted

2 way audio still not working
everything works perfectly with notifications, doorbell presses except for 2 way audio.

do we still need the onvif plugin for this?
i thought it should work natively with the reolink doorbell plugin

**Comment by Oakwhisper** [comment]
I was noticing that the [documentation](https://docs.scrypted.app/add-camera.html#reolink) indicates support for two way audio via the Reolink plugin, but the [NPM package](https://www.npmjs.com/package/@scrypted/reolink) indicates that it doesn't

**Reply by krizzziz** [reply]
Its says that it's not supported for the Reolink cameras. For the doorbell:

`The Reolink Doorbell supports two way audio via ONVIF. Reolink Cameras do not support this feature.`

**Reply by GravityRZ** [reply]
I want to read. WHAT IS SUPPORTED WITH THE REOLINK PLUGIN. 
I do not want to read what is supported with other plugins.

**Reply by koush** [reply, marked as answer]
two way audio works in the reolink plugin only for reolink doorbells. you must enable the checkbox that designates it a reolink doorbell. remove from homekit and then add it again.

**Reply by GravityRZ** [reply]
removed. reset plugin, added again, still no 2 way audio.

is this supposed to work when there is no detection or should it work always?
e.g press the talk button and you hear your voice coming out of the doorbell

**Reply by koush** [reply]
always works

**Reply by fortiko** [reply]
Hello @koush and thank you so much for your Scrypted

I also just removed my Reolink doorbell from Home, then re-added it, restarted the HomeKit plugin, and I still get the following error message when I press the "Talk" button in Home.app:

```

rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 2
Date: Wed, Nov 08 2023 13:26:52 GMT
Content-Base: rtsp://192.168.68.131/Preview_01_main/
Content-Type: application/sdp
Content-Length: 748

ONVIF Backchannel SDP:
v=0
o=- 1699434033733888 1 IN IP4 192.168.68.131
s=Session streamed by "preview"
i=Preview_01_main
t=0 0
a=tool:BC Streaming Media v2022.02.07
a=type:broadcast
a=control:*
a=range:npt=now-
a=x-qt-text-nam:Session streamed by "preview"
a=x-qt-text-inf:Preview_01_main
m=video 0 RTP/AVP 96
c=IN IP4 0.0.0.0
b=AS:500
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=640033;sprop-parameter-sets=Z2QAM6wVFKAoALWQ,aO48sA==
a=framerate:25.000
a=control:trackID=1
m=audio 0 RTP/AVP 97
c=IN IP4 0.0.0.0
b=AS:256
a=rtpmap:97 MPEG4-GENERIC/16000
a=fmtp:97 streamtype=5;profile-level-id=15;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3;config=1408; profile=1;
a=framerate:25.000
a=control:trackID=2

[HomeKit]: intercom failed to start RPCResultError: ONVIF audio backchannel not found
[HomeKit]:     at t.OnvifIntercom.checkIntercom (/onvif/src/onvif-intercom.ts:40:19)
[HomeKit]:     at process.intercomClient (node:internal/process/task_queues:95:5)
[HomeKit]:     at t.OnvifIntercom.startIntercom (/onvif/src/onvif-intercom.ts:50:35)
[HomeKit]:     at RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/src/rpc.ts:748:37)
[HomeKit]: @scrypted/reolink:host
[HomeKit]: host:@scrypted/homekit {
[HomeKit]:   cause: undefined
[HomeKit]: }
```

The connection from Scrypted to the doorbell is configured directly via the doorbell's IP, not through the NVR. Doorbell checkbox is enabled. 

2-way audio works fine with the Reolink app, as expected.

Is the error message "ONVIF audio backchannel not found" related to me using the Reolink plugin for this? From the discussion above, I understand that 2-way audio works with the Doorbell and the Reolink plugin, no need for ONVIF. In any case, ONVIF is enabled on the doorbell, I even disabled and re-enabled in case the setting was not applied. No change, non 2-way-audio for doorbell via Reolink plugin.

Any help would be welcome.

PS: Observation 1: I have several other (normal) Reolink cameras that I added via the Reolink NVR as different channels. Video works great. However, I can see that those cameras also have a "Talk" button, despite the Doorbell checkbox in Scrypted not being checked for those other (normal) cameras. So something seems to be broken at least in the logic of when to expose the Talk functionality, as it now shows for Doorbell with checkbox marked and normal cameras with checkbox unchecked.

PPS: Observation 2: When I press "Talk" on the Home.app on iPhone, I can hear myself faintly via the app itself with a few milliseconds delay, as if the audio is actually transmitted to the doorbell. Yet, at the doorbell itself no audio emits from the speaker. The speaker is known good however, as mentioned above, because the Reolink app has working 2-way-audio. Almost seems like the doorbell speaker needs to be activated somehow before 2-way-audio output to the doorbell works.

**Reply by fortiko** [reply]
Some more info, in case anybody is following: I installed the latest BETA firmware W.2555_23080700 for the doorbell from here: https://www.reddit.com/r/reolinkcam/comments/15h2kar/testers_wanted_for_doorbell_camera_new_firmware/

(BTW; I am using a WIFI doorbell, in case there is a difference to the PoE camera…)

After successful upgrade, I reviewed the camera "server" settings and realised that HTTPS was enabled, so I disabled HTTPS according to the documentation. I also enabled "Local Recording Stream" in the "Transcoding" plugin and manually selected Scrypted as the RTP sender in the "Homekit" plugin.

Result: working 2-way audio for a Reolink doorbell using the Reolink plugin.

I will hopefully be able to confirm if the BETA firmware is required by downgrading to stable W.2033_23041300, and will play around a bit with the settings around RTP sender and transcoding the local (i.e. iPhone or macOS) recording stream.

**Reply by fortiko** [reply]
Just to add why I enabled transcoding for the local recording stream: before I enabled it, I got messages similar to this:

```
rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 4
Date: Wed, Nov 08 2023 15:09:07 GMT
Range: npt=0.000-
Session: FE65DD66
RTP-Info: url=rtsp://192.168.68.111//track1;seq=0;rtptime=0,url=rtsp://192.168.68.111//track2;seq=0;rtptime=0,url=rtsp://192.168.68.111//track3;seq=0;rtptime=0

video codec/container not matched, transcoding: undefined {"url":"rtsp://127.0.0.1:36039","inputArguments":["-acodec","libopus","-i","rtsp://127.0.0.1:36039"]}
-hide_banner -acodec libopus -i rtsp:[REDACTED] -acodec pcm_mulaw -ar 8000 -ac 1 -ssrc -252886386 -dn -sn -vn -f rtp rtp://127.0.0.1:43312?pkt_size=1024 -sdp_file pipe:4
intercom playing
[rtsp @ 0x55c747710680] method SETUP failed: 461 Unsupported Transport
Input #0, rtsp, from 'rtsp://127.0.0.1:36039':
  Metadata:
    title           : HomeKit Audio Talkback
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Audio: opus, 48000 Hz, stereo, s16
Stream mapping:
  Stream #0:0 -> #0:0 (opus (libopus) -> pcm_mulaw (native))
Press [q] to stop, [?] for help

Output #0, rtp, to 'rtp://127.0.0.1:43312?pkt_size=1024':
  Metadata:
    title           : HomeKit Audio Talkback
    encoder         : Lavf58.76.100
  Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
    Metadata:
      encoder         : Lavc58.134.100 pcm_mulaw

video/audio detected, discarding further input

```

Since video and audio from the doorbell to HomeKit was working fine, I figured that the stream from Home.app to the doorbell must be the one that is failing. Not sure if that actually is the "local recording stream", but now it works ¯\_(ツ)_/¯

**Reply by GravityRZ** [reply]
Hello,
can you tell me where exactly you changed this
I do not have a transcoding plugin in scrypted
Also i can not finf the place to change the RTP sender in the homekit plugin.

EDIT FOUND IT:
Goto reolink doorbell plugin
goto transcoding
set local recording stream
goto homekit
set rtp sender to scrypted

tested it with my firmware but does not work

I also enabled "Local Recording Stream" in the "Transcoding" plugin and manually selected Scrypted as the RTP sender in the "Homekit" plugin.

**Reply by GravityRZ** [reply]
Hello,
can you tell me where exactly you changed this
I do not have a transcoding plugin in scrypted
Also i can not finf the place to change the RTP sender in the homekit plugin.

EDIT FOUND IT:
Goto reolink doorbell plugin
goto transcoding
set local recording stream
goto homekit
set rtp sender to scrypted

tested it with my firmware but does not work

I also enabled "Local Recording Stream" in the "Transcoding" plugin and manually selected Scrypted as the RTP sender in the "Homekit" plugin.

**Reply by fortiko** [reply]
@GravityRZ please check the following (in this order):

1. Disable HTTPS in the doorbell "server" settings and set up the other options as shown here: https://github.com/koush/scrypted/tree/e49f26b410f86e63b4487e4f8440e930cabfdb38/plugins/reolink
2.  Remove the doorbell from Home.app and Scrypted
3. Restart the Scrypted Reolink plugin
4. Add the doorbell again as a new doorbell with the same IP that you use to access the doorbell's web interface, and also make sure to check the "Doorbell" checkbox on the Reolink plugin general tab ("This camera is a Reolink Doorbell.")
5. Pair the doorbell again to Home.app as a standalone accessory with the QR code on the HomeKit plugin Pairing tab.
6. For good measure, force quit (kick out) the Home app and open it again. There is a 3-4 second delay for the doorbell audio, and sometimes you have to try two or three times to get talkback working. Be patient and methodical 👍 

Enclosed screenshots for all my configuration settings, I hope this helps.

Worst case, update your doorbell with the FW linked on the reddit post I linked above, and try again from Step 1.

<img width="424" alt="Screenshot 2023-11-08 at 21 03 42" src="https://github.com/koush/scrypted/assets/1612534/c9269bab-d0bc-4ed4-afb1-c13cef97757b">
<img width="453" alt="Screenshot 2023-11-08 at 21 03 56" src="https://github.com/koush/scrypted/assets/1612534/a36e24b9-9942-4020-85b7-ae9422bb5d20">
<img width="448" alt="Screenshot 2023-11-08 at 21 04 05" src="https://github.com/koush/scrypted/assets/1612534/47a3de81-00da-4d0f-b56c-a98318b0ab38">
<img width="453" alt="Screenshot 2023-11-08 at 21 04 14" src="https://github.com/koush/scrypted/assets/1612534/fcf814c1-bfa5-4738-ab49-f2f653256bf3">
<img width="446" alt="Screenshot 2023-11-08 at 21 04 22" src="https://github.com/koush/scrypted/assets/1612534/d6735c1b-8928-4cd1-9af4-53f999616092">
<img width="451" alt="Screenshot 2023-11-08 at 21 04 35" src="https://github.com/koush/scrypted/assets/1612534/29cca6e1-45c6-4dd7-b4ba-f0469d78a88f">
<img width="453" alt="Screenshot 2023-11-08 at 21 04 42" src="https://github.com/koush/scrypted/assets/1612534/c077425f-91f6-4b6b-b233-27bd7cf2d58f">

**Reply by GravityRZ** [reply]
I tried it but it still does not work
i have the latest firmware inside v3.0.0.2033_23041300

question regarding pairing.
do you have it paired with bridgmode or standalone accessory mode?

i use the standalone mode

i see this in the log maybee that shows where the problem is

![image](https://github.com/koush/scrypted/assets/16853535/2e69d1c4-a58d-432f-850f-9532e8777803)
![image](https://github.com/koush/scrypted/assets/16853535/61a8d3f9-ff20-4587-9682-8e8376bd7e78)
![image](https://github.com/koush/scrypted/assets/16853535/83f227ae-f7e5-48d2-9464-0763fd117137)

**Reply by GravityRZ** [reply]
I discovered the following.
when you open the camera from homekit the first time and press the talk button the camera does not react(no blue light)
when you press the talk button again(to stop talking)
close the camera window from homekit
open it again from homekit
press the talk button
NOW IT WORKS

it looks like it does not work the first time when opening the camera.

Do we need some kind of initialize function so it works every time

**Comment by GravityRZ** [comment]
i changed some settings and it seems to be working for now but i am testing if it works EVERY TIME
it now has the following behaviour

select camera from homekit (blue light stays off)
press talk button (blue light comes on)
press talk button again to stop talking(blue light stays on)
close live feed(blue button goes out.

i checked the log and these are the only errors i have at the moment.
@koush 
are you able to solve these errors, they problably are overruled but it is better to avoid them.

```
2023-11-11T10:59:41.647645961Z	stdout	[rtsp @ 0x55f8efd26b80] method SETUP failed: 461 Unsupported Transport
2023-11-11T10:59:41.352688647Z	stdout	
2023-11-11T10:59:41.352492175Z	stdout	intercom playing
2023-11-11T10:59:41.343274971Z	stdout	
2023-11-11T10:59:41.343185598Z	stdout	-hide_banner -acodec libopus -i rtsp:[REDACTED] -acodec pcm_alaw -ar 8000 -ac 1 -payload_type 8 -ssrc -1689903801 -dn -sn -vn -f rtp rtp://127.0.0.1:49420?pkt_size=1024 -sdp_file pipe:4
2023-11-11T10:59:41.342252856Z	stdout	
2023-11-11T10:59:41.342144579Z	stdout	video codec/container not matched, transcoding: undefined {"url":"rtsp://127.0.0.1:41692","inputArguments":["-acodec","libopus","-i","rtsp://127.0.0.1:41692"]}
2023-11-11T10:59:41.341703768Z	stdout	
```

**Comment by georgbachmann** [comment]
@GravityRZ could you tell me which settings you changed? Cause I can't get mine to work. I still have the official (not beta) firmware.
And did you add it via onvif or the reolink camera plugin?

**Reply by GravityRZ** [reply]
Sure,
i use the reolink plugin. i also have the onvif plugin installed but i have not camera's configured in that plugin.
please note that the first time you press the talk button it either does not work(no blue light) or it takes a little longer for the blue light to switch on
after that it works
also note that when you press the button again to stop talking the blue light stays on. it will only switch off when you kill the window

with the reolink app clicking the buton will switch on/off the light

[Reolink settings for Homekit with 2 way audio working.docx](https://github.com/koush/scrypted/files/14283557/Reolink.settings.for.Homekit.with.2.way.audio.working.docx)

**Reply by georgbachmann** [reply]
Ahhh... ok... that's true... if in homekit i press the "speak" button, then stop and then speak again, it works. If I don't do that, there is a suuuuuuuuper long delay... but that seems to work...
thanks for that hint... could have tried that myself :D thx!

**Reply by nin9s** [reply]
@koush I‘m observing the same behavior meaning that I have to click „talk“, click again to switch it off and re-enable - after that it seems to work. Are you able to fix that ? 
it’s always working with the native Reolink app after the first „talk“ enable

**Reply by benjaml4** [reply]
I'm having problems with two-way audio not working too... tried downgrading firmware and back to the latest firmware.

When 'Talk' button in the Home app on an iPhone is pressed the Reolink doorbell status light doesn't change to blue and no audio comes out from the doorbell speaker.

Would really like to get this working so tell me if you need more info from my side. Thanks! 🙏

**Log:**

```
[HomeKit]: addresses {
[HomeKit]:   sourceAddress: '192.168.1.3',
[HomeKit]:   sourceVideoPort: 53325,
[HomeKit]:   sourceAudioPort: 44031,
[HomeKit]:   targetAddress: '192.168.1.156',
[HomeKit]:   targetVidioPort: 61934,
[HomeKit]:   targetAudioPort: 52514
[HomeKit]: }
[HomeKit]: handleStreamRequest {
[HomeKit]:   sessionID: '50ad5558-6766-4cfe-bf3e-4f3a5bbaff9a',
[HomeKit]:   type: 'start',
[HomeKit]:   video: {
[HomeKit]:     codec: 0,
[HomeKit]:     profile: 1,
[HomeKit]:     level: 2,
[HomeKit]:     packetizationMode: 0,
[HomeKit]:     cvoId: undefined,
[HomeKit]:     width: 1280,
[HomeKit]:     height: 720,
[HomeKit]:     fps: 30,
[HomeKit]:     pt: 99,
[HomeKit]:     ssrc: 2691001335,
[HomeKit]:     max_bit_rate: 299,
[HomeKit]:     rtcp_interval: 0.5,
[HomeKit]:     mtu: 1378
[HomeKit]:   },
[HomeKit]:   audio: {
[HomeKit]:     codec: 'OPUS',
[HomeKit]:     channel: 1,
[HomeKit]:     bit_rate: 0,
[HomeKit]:     sample_rate: 24,
[HomeKit]:     packet_time: 20,
[HomeKit]:     pt: 110,
[HomeKit]:     ssrc: 3151458039,
[HomeKit]:     max_bit_rate: 24,
[HomeKit]:     rtcp_interval: 5,
[HomeKit]:     comfort_pt: 13,
[HomeKit]:     comfortNoiseEnabled: false
[HomeKit]:   }
[HomeKit]: }
[HomeKit]: Will wait for initial RTCP packet. { isHomeHub: undefined, isLowBandwidth: false, hasHomeHub: false }
[HomeKit]: { isLowBandwidth: false, isWatch: false, destination: 'local' }
[Rebroadcast Plugin]: Selected stream RTMP main.bcs
[Rebroadcast Plugin]: sending prebuffer 1993.5
[HomeKit]: { tool: undefined, rtpSender: 'Scrypted' }
[HomeKit]: Waiting for video RTCP packet before sending video.
[HomeKit]: packet loss 1
[HomeKit]: Received first video RTCP packet.
[HomeKit]: rtsp/tcp h264
[HomeKit]: a/v h264 opus found h264 undefined
[HomeKit]: rtsp/tcp aac
[Rebroadcast Plugin]: RTMP main.bcs active rebroadcast clients: 1
[HomeKit]: -hide_banner -analyzeduration 0 -probesize 512 -i rtsp:[REDACTED] -acodec libopus -application lowdelay -frame_duration 20 -flags +global_header -ar 24k -b:a 24k -bufsize 96k -ac 1 -payload_type 110 -ssrc 15617258 -dn -sn -vn -f rtp rtp://127.0.0.1:35784?pkt_size=400 -sdp_file pipe:4
[HomeKit]: received first video packet
[HomeKit]: scanning for idr start found: 28 1
[HomeKit]: scanning for idr start found: 28 1 24 7 8
[HomeKit]: idr start found: 28 5
[HomeKit]: [rtsp @ 0x55ba45b6b0] method SETUP failed: 461 Unsupported Transport
[HomeKit]: Input #0, rtsp, from 'rtsp://127.0.0.1:40383':
[HomeKit]:   Metadata:
[HomeKit]:     title           : No Name
[HomeKit]:   Duration: N/A, start: 0.000000, bitrate: N/A
[HomeKit]:   Stream #0:0: Audio: aac (LC), 16000 Hz, mono, fltp
[HomeKit]: Stream mapping:
[HomeKit]:   Stream #0:0 -> #0:0 (aac (native) -> opus (libopus))
[HomeKit]: Press [q] to stop, [?] for help
[HomeKit]: Output #0, rtp, to 'rtp://127.0.0.1:35784?pkt_size=400':
[HomeKit]:   Metadata:
[HomeKit]:     title           : No Name
[HomeKit]:     encoder         : Lavf58.76.100
[HomeKit]:   Stream #0:0: Audio: opus, 24000 Hz, mono, flt, 24 kb/s
[HomeKit]:     Metadata:
[HomeKit]:       encoder         : Lavc58.134.100 libopus
[libopus @ 0x55ba4681c0] Queue input is backward in time.2kbits/s speed=3.75e+04x    
[HomeKit]: [rtp @ 0x55ba46b9c0] Non-monotonous DTS in output stream 0:0; previous: 3720, current: 1608; changing to 3720. This may result in incorrect timestamps in the output file.
[HomeKit]: [rtp @ 0x55ba46b9c0] Non-monotonous DTS in output stream 0:0; previous: 3720, current: 2568; changing to 3720. This may result in incorrect timestamps in the output file.
[HomeKit]: video/audio detected, discarding further input
[HomeKit]: received first audio packet
rtsp outgoing message
 OPTIONS rtsp://192.168.1.37:554/ RTSP/1.0
CSeq: 0
User-Agent: Scrypted



rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 0
Date: Wed, Apr 10 2024 23:09:39 GMT
Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, PAUSE, GET_PARAMETER, SET_PARAMETER

rtsp outgoing message
 DESCRIBE rtsp://192.168.1.37:554/ RTSP/1.0
Require: www.onvif.org/ver20/backchannel
Accept: application/sdp
CSeq: 1
User-Agent: Scrypted



rtsp incoming message
 RTSP/1.0 401 Unauthorized
CSeq: 1
Date: Wed, Apr 10 2024 23:09:39 GMT
WWW-Authenticate: Digest realm="BC Streaming Media", nonce="d1b35ceb78cc91e0959ad131e3ca318f"

rtsp outgoing message
 DESCRIBE rtsp://192.168.1.37:554/ RTSP/1.0
Require: www.onvif.org/ver20/backchannel
Accept: application/sdp
CSeq: 2
User-Agent: Scrypted
Authorization: Digest username="admin", realm="BC Streaming Media", nonce="d1b35ceb78cc91e0959ad131e3ca318f", uri="rtsp://192.168.1.37:554/", algorithm="MD5", response="56e0001da5e96c7542e11521330c4aa4"



rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 2
Date: Wed, Apr 10 2024 23:09:39 GMT
Content-Base: rtsp://192.168.1.37//
Content-Type: application/sdp
Content-Length: 740

ONVIF Backchannel SDP:
v=0
o=- 1712788694674900 1 IN IP4 192.168.1.37
s=Session streamed by "preview"
t=0 0
a=tool:BC Streaming Media v202210012022.10.01
a=type:broadcast
a=control:*
a=range:npt=now-
a=x-qt-text-nam:Session streamed by "preview"
m=video 0 RTP/AVP 96
c=IN IP4 0.0.0.0
b=AS:8192
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=640033;sprop-parameter-sets=Z2QAM6wVFKAoAPGQ,aO48sA==
a=recvonly
a=control:track1
m=audio 0 RTP/AVP 97
c=IN IP4 0.0.0.0
b=AS:8192
a=rtpmap:97 MPEG4-GENERIC/16000
a=fmtp:97 streamtype=5;profile-level-id=1;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3;config=1408
a=recvonly
a=control:track2
m=audio 0 RTP/AVP 0
a=control:track3
a=rtpmap:0 PCMU/8000
a=sendonly
rtsp outgoing message
 SETUP rtsp://192.168.1.37//track3 RTSP/1.0
Require: www.onvif.org/ver20/backchannel
Transport: RTP/AVP;unicast;client_port=38225-38226
CSeq: 3
User-Agent: Scrypted
Authorization: Digest username="admin", realm="BC Streaming Media", nonce="d1b35ceb78cc91e0959ad131e3ca318f", uri="rtsp://192.168.1.37//track3", algorithm="MD5", response="b587df700a366e6661391e384550ab1e"



rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 3
Date: Wed, Apr 10 2024 23:09:39 GMT
Transport: RTP/AVP;unicast;destination=192.168.1.3;source=192.168.1.37;client_port=38225-38226;server_port=6970-6971
Session: 6778772E;timeout=65

backchannel transport {
  'RTP/AVP': undefined,
  unicast: undefined,
  destination: '192.168.1.3',
  source: '192.168.1.37',
  client_port: '38225-38226',
  server_port: '6970-6971'
}
rtsp outgoing message
 PLAY rtsp://192.168.1.37:554/ RTSP/1.0
Require: www.onvif.org/ver20/backchannel
Range: npt=0.000-
CSeq: 4
User-Agent: Scrypted
Authorization: Digest username="admin", realm="BC Streaming Media", nonce="d1b35ceb78cc91e0959ad131e3ca318f", uri="rtsp://192.168.1.37:554/", algorithm="MD5", response="77387e617dbc8f87a813397b7c9a86b3"
Session: 6778772E



rtsp incoming message
 RTSP/1.0 200 OK
CSeq: 4
Date: Wed, Apr 10 2024 23:09:39 GMT
Range: npt=0.000-
Session: 6778772E
RTP-Info: url=rtsp://192.168.1.37//track1;seq=0;rtptime=0,url=rtsp://192.168.1.37//track2;seq=0;rtptime=0,url=rtsp://192.168.1.37//track3;seq=0;rtptime=0

video codec/container not matched, transcoding: undefined {"url":"rtsp://127.0.0.1:32993","inputArguments":["-acodec","libopus","-i","rtsp://127.0.0.1:32993"]}
-hide_banner -acodec libopus -i rtsp:[REDACTED] -acodec pcm_mulaw -ar 8000 -b:a 64k -ac 1 -ssrc 1847012146 -dn -sn -vn -f rtp rtp://127.0.0.1:40807?pkt_size=1024 -sdp_file pipe:4
intercom playing
[rtsp @ 0x55a80ba500] method SETUP failed: 461 Unsupported Transport

Input #0, rtsp, from 'rtsp://127.0.0.1:32993':

  Metadata:
    title           : HomeKit Audio Talkback
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Audio: opus, 48000 Hz, stereo, s16

Stream mapping:
  Stream #0:0 -> #0:0 (opus (libopus) -> pcm_mulaw (native))
Press [q] to stop, [?] for help

Output #0, rtp, to 'rtp://127.0.0.1:40807?pkt_size=1024':
  Metadata:
    title           : HomeKit Audio Talkback
    encoder         : Lavf58.76.100
  Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
    Metadata:
      encoder         : Lavc58.134.100 pcm_mulaw
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x    
video/audio detected, discarding further input
[HomeKit]: handleStreamRequest { sessionID: '50ad5558-6766-4cfe-bf3e-4f3a5bbaff9a', type: 'stop' }
[HomeKit]: streaming session killed, duration: 9s
[HomeKit]: Stopping intercom.
[Rebroadcast Plugin]: RTMP main.bcs active rebroadcast clients: 0
rtsp outgoing message
 TEARDOWN rtsp://192.168.1.37:554/ RTSP/1.0
CSeq: 5
User-Agent: Scrypted
Authorization: Digest username="admin", realm="BC Streaming Media", nonce="d1b35ceb78cc91e0959ad131e3ca318f", uri="rtsp://192.168.1.37:554/", algorithm="MD5", response="ce4b8a0ac4ca9abc84806d367dbdbf2f"
Session: 6778772E



ffmpeg exited
[HomeKit]: ffmpeg exited
```

**Comment by cat2devnull** [comment]
Just an update on this topic. I was able to get things working with the following;
- Firmware: Requires v3.0.0.2033_23041302 (newer v3.0.0.3215_2401262241 is broken) [Issue 1376](https://github.com/koush/scrypted/issues/1376)
- Username/password: Avoid special chars as this breaks auth, as per [Issue 879](https://github.com/koush/scrypted/issues/879) and make sure the user has admin privileges.
- Plugin Version: Avoid Reolink Plugin v0.6.3/0.6.4 as per [Issue 1405](https://github.com/koush/scrypted/issues/1405)
- Transcoding: Not required
- Streams and Snapshots: Can just use defaults
- HomeKit: Use standalone accessory mode and don't forget to restart the HomeKit module after adding the doorbell
- and don't forget to tick the "Doorbell" checkbox
Hope this helps anyone that was struggling like me (I had an incompatible firmware that took me a while to work out).

**Reply by krizzziz** [reply]
Thanks for this! Seems to be working for me as well. Only problem is that it stops working after a certain time. And I can’t figure out why.

**Reply by nin9s** [reply]
But it’s only working for a short amount of time unfortunately

**Reply by paolotremadio** [reply]
How long until it stops working? Are we talking about days or weeks? Also, has anyone found a solution to the 1-3s lags on the 2-way audio? (talking back from HomeKit to the doorbell)

**Comment by magic2910** [comment]
Where can I get v3.0.0.2033_23041302 ? Have the newest installed :(

**Reply by cat2devnull** [reply]
Easiest way to get old firmwares is here; https://github.com/AT0myks/reolink-fw-archive/releases
To save you the trouble;
POE v3.0.0.2033_23041302 - https://home-cdn.reolink.us/wp-content/uploads/2023/05/080159231683511163.4286.zip

**Reply by magic2910** [reply]
Thank you. Now 2 way Audio is working

**Reply by soerensteen** [reply]
After I installed that old firmware, my chime no longer works and I cannot re pair it. Two-way audio works though 😂

**Reply by KumarMudedla** [reply]
My Doorbell POE bricked after this firmware downgrade. Returning and getting new one today. Any one had this issue ? its V2 Doorbell. also Is the Chime working after downgrade ?

**Reply by KumarMudedla** [reply]
adding for any one coming here.. Reolink Support Sent me Newer version '3308_2407041199' that resolving the 2 way audio issue on V2 and also Chime still works with this version

**Reply by E-Coll** [reply]
> My Doorbell POE bricked after this firmware downgrade. Returning and getting new one today. Any one had this issue ? its V2 Doorbell. also Is the Chime working after downgrade ?

@KumarMudedla 

I bricked my Camera too, I can't recommend downgrading to the old firmware at the moment :)
I downgraded with MacOS Reolink Software

**Reply by E-Coll** [reply]
> adding for any one coming here.. Reolink Support Sent me Newer version '3308_2407041199' that resolving the 2 way audio issue on V2 and also Chime still works with this version

How were you able to fix your brick?

**Reply by magic2910** [reply]
Could you Upload the 1199 version?

**Reply by KumarMudedla** [reply]
I was not able fix the brick, got a new one and then reolunk support sent new fireware in the email

**Reply by cbuxton5200** [reply]
@KumarMudedla It doesn't make sense that there are different firmwares for black vs white?  I would have guessed that the firmware with the "P" should have been for the PoE version and the file with the "W" should have been the wifi version.  Do you have any idea if those are just for PoE?

**Reply by JuergenLeber** [reply]
@cbuxton5200 There are indeed different PAK files for white and black. The white wifi version has for example the hardware ID DB_566128M5MP_W_W. And yes, I'm also waiting for the wifi version for the white camera but Reolink support somehow doesn't want to provide it.

**Reply by paolotremadio** [reply]
I'm also looking for the DB_566128M5MP_W_W firmware :|

**Comment by Midifreakz** [comment]
Hi @cat2devnull Thank you for the links. This firmware indeed enables 2 way audio. Unfortunately it breaks the ability to send pre configured auto reply messages over REST commands. :(
